sdk/iot/inc/az_iot_hub_properties.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_span.h>
+#include <az_result.h>
+
+#ifndef _az_IOT_HUB_PROPERTIES_H
+#define _az_IOT_HUB_PROPERTIES_H
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_iot_hub_properties {
+    struct {
+        az_span property_string;   // "URIENCODED(name1=val1&name2=val2&...)"
+    } _internal;
+} az_iot_hub_properties;
+
+// TODO: Do we need an enumerator (requires state preserved)
+
+az_result az_iot_hub_properties_add(az_iot_hub_properties *properties, az_span name, az_span value);

```

Comment:
>az_iot_hub_properties *properties [](start = 36, length = 33)

Do we need this new type, az_span is enough?

Reply:
The reason is an idea from @danewalton that was also confirmed by @JeffreyRichter : the properties is very similar to an URI query string `name1=value1&name2=value2&...&namen=valuen`: the type definition above (the internal portion) will change to accommodate the CRUD operations and potentially an enumerator.
We recommend that the property list will not normally change, especially during send and so the buffers of previous messages can be reused with minimal changes. (Also, we would recommend that the last tuple changes if necessary as it won't require any major string editing.)

For details on what I mean, see az_http_request's query parsing using query_start: https://github.com/Azure/azure-sdk-for-c/blob/25f8a0228e5f250c02e389f19d88c064c93959c1/sdk/core/core/inc/az_http.h#L44

@JeffreyRichter may have a better URI example.